runtime/src/accounts_db.rs

```diff
@@ -2209,67 +2251,186 @@ impl AccountsDb {
+                        // 1) Shrink has removed the old storage entry and rewritten to
+                        // a newer storage entry
+                        // 2) The `pubkey` asked for in this function is a zero-lamport account,
+                        // and the storage entry holding this account qualified for zero-lamport clean.
+                        //
+                        // In both these cases, it should be safe to retry and recheck the accounts
+                        // index. Also note that in both cases, if we do find the storage entry,
+                        // we can guarantee that the storage entry is safe to read from because
+                        // we grabbed a reference to the storage entry while it was still in the
+                        // storage map. This means even if the storage entry is removed from the storage
+                        // map after we grabbed the storage entry, the recycler should not reset the
+                        // storage entry until we drop the reference to the storage entry.
+
+                        // RPC get_account() may have fetched an old root from the index that was
+                        // either:
+                        // 1) Cleaned up by clean_accounts(), so the accounts index has been updated
+                        // and the storage entries have been removed.
+                        // 2) Dropped by purge_slots() because the slot was on a minor fork, which
+                        // removes the slots' storage entries but doesn't purge from the accounts index
+                        // (account index cleanup is left to clean for stored slots).
+                        num_acceptable_failed_iterations += 1;
+                    }
+                }
+                LoadedAccountAccessor::Cached(None) => {
+                    // Cache was flushed in between checking the index and retrieving from the cache,
+                    // so retry.
+                    num_acceptable_failed_iterations += 1;
+                    if is_root_fixed {
+                        // it's impossible for this to fail for transaction loads more than once.
+                        // This is because:
+                        // 1) For a slot `X` that's being replayed, there is only one
+                        // latest ancestor containing the latest update for the account, and this
+                        // ancestor can only be flushed once.
+                        // 2) The root cannot move while replaying, so the index cannot continually
+                        // find more up to date entries than the current `slot`
+                        assert!(num_acceptable_failed_iterations <= 1);
+                    }
+                }
+                LoadedAccountAccessor::Cached(Some(_)) | LoadedAccountAccessor::Stored(Some(_)) => {
+                    // Everything else means there was no race, so break out and continue
+                    break;
```

Comment:
To realize this: https://github.com/solana-labs/solana/pull/15454/files#r590769646, maybe this `break` can simply replaced with `return Some((account_accessor, slot))`. which I think more readable in this case.

Reply:
that's because I don't think we should generally mix `break` and `return` inside a loop.. xD

Reply:
let's stick to `return` for this case. :)